controller/user-controller.js

```diff
@@ -0,0 +1,116 @@
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const db = require("../db/db");
+require("dotenv").config();
+const { createUsersAccount } = require("../db/queries/set-up-user-table");
+
+// create user table for testing purposes
+createUsersAccount();
+
+const createUser = async (req, res) => {
+  const user = req.user;
+  const userData = req.body;
+
+  // Check if the authenticated user is an admin
+  if (user.role !== "admin") {
+    return res.status(403).json({
+      status: "error",
+      error: "Only admin users can create account",
+    });
+  }
+
+  // Check if email already exists
+  const emailCheckQuery = "SELECT COUNT(*) FROM users WHERE email = $1";
+  const emailCheckValues = [userData.email];
+  const emailCheckResult = await db.query(emailCheckQuery, emailCheckValues);
+  const emailExists = emailCheckResult.rows[0].count > 0;
+
+  if (emailExists) {
+    return res.status(400).json({
+      status: "error",
+      error: "Email already exists",
+    });
+  }
+
+  const hashedPassword = await bcrypt.hash(userData.password, 10);
+  const insertUserDataQuery = `
+      INSERT INTO users (
+        firstName, lastName, email, password, gender, job_role, department, address, role, created_on
+      ) VALUES (
+        $1, $2, $3, $4, $5, $6, $7, $8, $9, $10
+      ) RETURNING *;             
+    `;
+  const values = [
+    userData.firstName,
+    userData.lastName,
+    userData.email,
+    hashedPassword,
+    userData.gender,
+    userData.job_role,
+    userData.department,
+    userData.address,
+    userData.role,
+    new Date(),
+  ];
+
+  try {
+    const result = await db.query(insertUserDataQuery, values);
+    const createdUser = result.rows[0];
+
+    // Create a payload for token
+    const tokenPayload = {
+      userId: createdUser.id,
+      email: createdUser.email,
+      role: createdUser.role,
+    };
+
+    // Sign the payload with the secret key and generate the token
+    const token = jwt.sign(tokenPayload, process.env.JWT_SECRET, {
+      expiresIn: "24h",
+    });
```

Comment:
You Don't Need a Token After Creating a User, We Only Need It After Login

Reply:
yes but the response in the pdf wants me to return token

Reply:
after user is created

Reply:
Have removed it thou